src/KafkaW/ConsumerFactory.cpp

```diff
@@ -1,15 +1,27 @@
 #include "ConsumerFactory.h"
+#include "helper.h"
 
 namespace KafkaW {
 
-std::unique_ptr<Consumer> createConsumer(const BrokerSettings &Settings) {
-  std::string ErrorString;
+std::unique_ptr<Consumer> createConsumer(const BrokerSettings &Settings,
```

Comment:
It took me longer than I wanted to wrap my head around what is happening here but after having done so, I determined that there is a design problem that we should fix seeing as we are touching the code anyways.

The `BrokerSettings` class should not be responsible for applying any configuration to a `RdKafka::Conf` object. This should either be done in this function (too ugly), by the `RdKafka::Conf` object itself (too hard) or by a free function (preferred).

Reply:
Oh, and the function that implements this functionality should **not** be called `apply`.